models/baseModels/Invoice/Invoice.ts

```diff
@@ -1047,19 +1047,16 @@ export abstract class Invoice extends Transactional {
           if (sinvreturnedDoc.outstandingAmount?.isZero()) {
             return this.grandTotal?.abs();
           } else {
-            const totalPaid = sinvreturnedDoc.grandTotal
-              ?.abs()
-              .sub(sinvreturnedDoc.outstandingAmount as Money);
+            const totalPaid = sinvreturnedDoc
+              .grandTotal!.abs()
+              .sub(sinvreturnedDoc.outstandingAmount!);
 
-            if (!sinvreturnedDoc.isReturn) {
-              return totalPaid?.abs();
-            }
-
-            if (!this.grandTotal?.isNegative()) {
-              this.grandTotal = this.grandTotal?.neg();
-            }
+            const returnAmount = this.grandTotal!.abs();
 
-            return this.grandTotal?.add(totalPaid as Money).abs();
+            const result = returnAmount.lte(totalPaid)
+              ? returnAmount
+              : totalPaid;
+            return result;
```

Comment:
```suggestion
            return returnAmount.lte(totalPaid)
              ? returnAmount
              : totalPaid;
```

You can directly return this and use a more meaningful variable name.